src/query/query_hash_scan.c

```diff
@@ -2570,9 +2585,17 @@ fhs_insert_to_dk_bucket (THREAD_ENTRY * thread_p, FHSID * fhsid_p, VPID * next_b
       goto error;
     }
 
-  if (dk_bucket_page_p)
+  if (last_dk_bucket_page_p == first_dk_bucket_page_p)
```

Comment:
Codes after the 2588 line appear to have the same content around the "error:" label.

If you add a variable to contain the return value, it will be more concise.

Reply:
Okay. I fixed it.